data-prepper-plugins/kafka-source/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSourceBuffer.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.kafka.source;
+
+import io.micrometer.core.instrument.Counter;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeoutException;
+import org.apache.kafka.clients.consumer.CommitFailedException;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.log.JacksonLog;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.plugins.kafka.source.configuration.TopicConfig;
+import org.opensearch.dataprepper.plugins.kafka.source.util.MessageFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * * A helper utility class which helps to write different formats of records
+ * like json and string to the buffer.
+ */
+@SuppressWarnings("deprecation")
+public class KafkaSourceBuffer<K, V> {
```

Comment:
This call is more of a Util class and not actually a buffer class, can this be renamed appropriately ?

Reply:
Resolved.
This class is renamed as KafkaSourceBufferAccumulator.java